RF_PBMC.R

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env Rscript
+
+
+# Load libraries
+suppressPackageStartupMessages(library(optparse))
+suppressPackageStartupMessages(library(logging))
+suppressPackageStartupMessages(library(dplyr))
+suppressPackageStartupMessages(library(randomForest))
+suppressPackageStartupMessages(library(Seurat))
+
+ 
+# process_data function follows Seurat tutorial for processing PBMC data set made available by 10x Genomics
+# dataset https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz
+# Seurat tutorial https://satijalab.org/seurat/pbmc3k_tutorial.html
+process_data <- function(input.data){
+	dense.size <- object.size(x = as.matrix(x = input.data))
+	print(dense.size)
+	seuratObj <- CreateSeuratObject(raw.data = input.data, min.cells = 3, min.genes = 200, project = "10X_data")
+	print(seuratObj)
+	mito.genes <- grep(pattern = "^MT-", x = rownames(x = seuratObj@data), value = TRUE)
+	percent.mito <- Matrix::colSums(seuratObj@raw.data[mito.genes, ])/Matrix::colSums(seuratObj@raw.data)
+	seuratObj <- AddMetaData(object = seuratObj, metadata = percent.mito, col.name = "percent.mito")
+	vln <- VlnPlot(object = seuratObj, features.plot = c("nGene", "nUMI", "percent.mito"), nCol = 3)
+	print(vln)
+	par(mfrow = c(1, 2))
+	GenePlot(object = seuratObj, gene1 = "nUMI", gene2 = "percent.mito")
+	GenePlot(object = seuratObj, gene1 = "nUMI", gene2 = "nGene")
+	seuratObj <- FilterCells(object = seuratObj, subset.names = c("nGene", "percent.mito"), low.thresholds = c(200, -Inf), high.thresholds = c(2500, 0.05))
+	seuratObj <- NormalizeData(object = seuratObj, normalization.method = "LogNormalize", scale.factor = 10000)
+	seuratObj <- FindVariableGenes(object = seuratObj, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)
+	length(x = seuratObj@var.genes)
```

Comment:
Remove this line or print out the genes in logging. Length is not being used.

Reply:
removed unneeded line